src/cluster_legacy.c

```diff
@@ -273,6 +283,54 @@ int auxHumanNodenamePresent(clusterNode *n) {
     return sdslen(n->human_nodename);
 }
 
+int auxClientIpV4Setter(clusterNode *n, void *value, int length) {
+    if (n && !strncmp(value, n->client_ipv4, length)) {
```

Comment:
It's checked in if. Do you want `n != NULL` instead if just `n`?

It's a matter of style. I can agree `n != NULL` is clearer, but more common seems to be `n`, at least in this file.

Reply:
It can't be null in human nodename setter either, right?

Reply:
Seems like it should have checked if `n->client_ipv4` is null instead, to avoid storing an empty string. Are empty strings stored just to be gossipped to clear a previous value thought the cluster?

Reply:
>Are empty strings stored just to be gossipped to clear a previous value thought the cluster?

I think the current convention (`hostname`, `human_nodename`, etc) uses an implicit model, which does require storing an empty string in order to clear the previous value but it does **NOT** involve sending empty ping extensions. In fact, an empty string will lead to the absence of the corresponding ping extension, which will be interpreted as a clear request in the `hostname` and `human_nodename` cases (`shard_id` is a bit different because a node must have a shard_id). This model works when we keep coupling the ping extension with the gossip message. Since these extensions are relatively small (compared to the 2KB slot bitmap), I think it is OK to continue with sending all extensions in all gossip messages, which helps with reliability and is also less expensive when these extensions are not enabled.

Reply:
Makes sense. I see now that we don't send the empty strings. We do store them though, when we could store NULL instead and save a few bytes. That can be improved in the future.